### PR TITLE
RES-335 add comments; add one single SNP test

### PR DESCRIPTION
**Pull request by yunfeiguo**
Hi @marghoob 

I added a few comments to the code for better maintainability and readability. Meanwhile, I created the first integration test: a single SNP simultation test for varsim. I want to submit a PR as early as possible to avoid late surprises. Thanks!

p.s. I can't assign reviewer/label/milestone on PR creation page, do you know why? Also, do we need to use some templates similar to seqalto?

### CODE REVIEW

File: src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java [hunk 1]
```diff
@@ -144,8 +178,23 @@ private void addVariant(ChrString chr, Variant var) {
...
+     * Each variant is decomposed into a combination of deletion
+     * and insertion. Original sequence is flagged at variant loci.
+     * Perturbed sequences will be generated based on these flags.
+     * Map file format (MFF) records will generated by iterating
+     * over flag array of each sequence. VCFs (one for each sequence)
+     * containing all sampled variants will be generated as well
+     * with optional phasing information (if both maternal and
+     * paternal genomes are chosen for output).
+     *
+     */
```
Comment (marghoob) [inline src/main/java/com/bina/varsim/tools/simulation/VCF2diploid.java]: 👍

**Comment by yunfeiguo** [comment]
@marghoob After I gave it a second thought, I think I'd better create a new branch for each PR. Sometimes I still want to make changes to the branch before it is waiting for approval. Thanks!

**Comment by marghoob** [comment]
I agree. Typically, you create a branch for an issue in JIRA and make a pull request. After the request gets merged, you delete the branch and move to another issue.

**Comment by yunfeiguo** [comment]
Cool. Thank you!

**Pull request merged**